### ISSUE TITLE
Host header is being overwritten with upstream name

### ISSUE THREAD

**Issue by lw346** [opentracing-contrib/nginx-opentracing#56]
If nginx is configured on host `my.host`, with a backend set up, `my-backend`, when I make a request to `my.host/some-path` with a header `Host: my.host`, then the `Host` header is being overwritten with `my-backend` when propagated to the upstream.

Here's an example failing test:

```
diff --git a/test/environment/app.py b/test/environment/app.py
index c190af6..d51546a 100644
--- a/test/environment/app.py
+++ b/test/environment/app.py
@@ -12,5 +12,11 @@ def has_span_context_redirect():
     assert 'x-ot-span-context' in request.headers
     return '', status.HTTP_301_MOVED_PERMANENTLY
 
+@app.route('/has-host-header')
+def has_host_header():
+    assert 'host' in request.headers
+    assert 'localhost' == request.headers['host']
+    return 'OK'
+
 if __name__ == '__main__':
     app.run(debug=True,host='0.0.0.0')
diff --git a/test/environment/nginx.conf b/test/environment/nginx.conf
index 0ee3ab8..67b72ff 100644
--- a/test/environment/nginx.conf
+++ b/test/environment/nginx.conf
@@ -20,6 +20,10 @@ http {
       proxy_pass http://backend/has-span-context;
     }
 
+    location = /has-host-header {
+      opentracing_propagate_context;
+      proxy_pass http://backend/has-host-header;
+    }
+
     location = /manual-propagation {
       proxy_set_header x-ot-span-context $opentracing_context_x_ot_span_context;
 
diff --git a/test/nginx_opentracing_test.py b/test/nginx_opentracing_test.py
index c7cb8c0..2929983 100644
--- a/test/nginx_opentracing_test.py
+++ b/test/nginx_opentracing_test.py
@@ -87,6 +87,15 @@ class NginxOpenTracingTest(unittest.TestCase):
         with open(os.path.join(self.workdir, "traces", "nginx.json")) as f:
             self.nginx_traces = json.load(f)
 
+    def testHostHeader(self):
+        self.conn.request("GET", "/has-host-header")
+        response = self.conn.getresponse()
+
+        self.assertEqual(response.status, 200)
+        self._stopEnvironment()
+
+        self.assertEqual(len(self.nginx_traces), 2)
+
     def testTrivialRequest(self):
         self.conn.request("GET", "/")
         response = self.conn.getresponse()
```

Having tested this behaviour in real nginx, it may also be stripping out `proxy_set_header` directives if defined in the `http` block.

**Comment by lw346** [comment]
It looks like when `opentracing_propagate_context` all previous config arguments are discarded and replaced with the new opentracing args exclusively.

**Comment by lw346** [comment]
Taking this down to more basic levels:

```
events {}

http {
  opentracing on;
  opentracing_load_tracer /usr/local/lib/libopentracing_mocktracer.so /tracer-config.json;

  proxy_set_header Host $host;

  upstream backend {
    server backend:5000;
  }

  server {
    error_log logs/debug.log debug;
    listen 8080;
    server_name localhost;

    location = /has-host-header {
      proxy_set_header my-header 'my-value';
      proxy_pass http://backend/has-host-header;
    }
}
```

This example also fails, even without opentracing turned on.  So I think being able to specify `opentracing_propagate_context` at a higher level (as per https://github.com/opentracing-contrib/nginx-opentracing/pull/57) is the fix.

**Comment by rnburn** [comment]
Fixed in #57
